pallets/ddc-customers/src/lib.rs

```diff
@@ -511,6 +515,26 @@ pub mod pallet {
 
 			Ok(())
 		}
+
+		/// Mark existing bucket with specified bucket id as removed
+		///
+		/// Only an owner can remove a bucket
+		#[pallet::call_index(6)]
+		// Todo: add benchmarks
+		#[pallet::weight(T::WeightInfo::create_bucket())]
+		pub fn remove_bucket(origin: OriginFor<T>, bucket_id: BucketId) -> DispatchResult {
+			let owner = ensure_signed(origin)?;
+			let mut bucket = Self::buckets(bucket_id).ok_or(Error::<T>::NoBucketWithId)?;
```

Comment:
Use [try_mutate](https://paritytech.github.io/polkadot-sdk/master/frame_support/storage/trait.StorageMap.html#tymethod.try_mutate) which is cleaner and a best practice.

Reply:
The idea behind it is that instead of using a get / change / insert (replace) pattern, we use `try_mutate` to get and mutate the object stored in the storage and it would only be stored if `return Ok` was returned by the lambda. We can perform the checks of already deleted and others inside the lambda.

Reply:
Updated